pkg/retention/config.go

```diff
@@ -0,0 +1,73 @@
+package retention
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	configmapName         = "k8s-backup-operator-retention"
+	defaultConfigFilePath = "/config/retention"
+	strategyKey           = "strategy"
+	DefaultStrategy       = KeepAllStrategy
+)
+
+var defaultConfig = Config{Strategy: DefaultStrategy}
+
+type Config struct {
+	Strategy StrategyId
+}
+
+type ConfigGetter struct {
+	configFilePath string
+}
+
+func NewConfigGetter() *ConfigGetter {
+	return &ConfigGetter{configFilePath: defaultConfigFilePath}
+}
+
+func (sg *ConfigGetter) GetConfig(ctx context.Context) (Config, error) {
```

Comment:
An alternative to reading the retention-strategy from a mounted config-map, could be to just set it as an program-argument in the "args" of the CronJob.
This way the retention-config (schedule and strategy) is in one place the CronJob.
Just an idea...

Reply:
Hm yes, we could do that (since we're using Helm values to set both anyway).

Reply:
Done, thanks for the great idea! This simplifies things!